simplified-notifications/src/main/java/NotificationTokenHTTPCalls.kt

```diff
@@ -35,61 +37,66 @@ class NotificationTokenHTTPCalls(
   override fun registerFCMTokenForProfileAccount(
     account: AccountType
   ) {
+    val credentials = account.loginState.credentials
+    val originalUrl = credentials?.deviceRegistrationURI
+
+    if (originalUrl == null) {
+      logger.debug("Account {} doesn't have a device registration uri", account.id)
+      return
+    }
+
     firebaseInstance.token
       .addOnSuccessListener { token ->
         logger.debug("Success fetching FCM Token: {}", token)
-        val originalUrl = buildString {
-          this.append(account.provider.catalogURI.toString())
-          this.append(URLEncoder.encode("patrons/me/devices", "utf-8"))
-        }
+
         val queryUrl = buildString {
           this.append(originalUrl)
           this.append("?")
           this.append("device_token=")
           this.append(URLEncoder.encode(token, "utf-8"))
         }
 
-        val credentials = account.loginState.credentials
-
         val request = this.http.newRequest(queryUrl)
           .setAuthorization(AccountAuthenticatedHTTP.createAuthorizationIfPresent(credentials))
           .addCredentialsToProperties(credentials)
           .build()
 
-        val response = request.execute()
-        when (val status = response.status) {
-          is LSHTTPResponseStatus.Responded.OK -> {
-            account.updateBasicTokenCredentials(status.getAccessToken())
-            logger.debug("The account {} has the FCM token {}", account.id, token)
-          }
+        this.executor.execute {
+          val response = request.execute()
+          when (val status = response.status) {
+            is LSHTTPResponseStatus.Responded.OK -> {
+              account.updateBasicTokenCredentials(status.getAccessToken())
+              logger.debug("The account {} has the FCM token {}", account.id, token)
+            }
 
-          is LSHTTPResponseStatus.Responded.Error -> {
-            if (status.properties.status == HttpURLConnection.HTTP_NOT_FOUND) {
-              logger.error(
-                "The account {} doesn't have the FCM token {}. Let's send it...",
-                account.id,
-                token
-              )
-              setFCMTokenForAccount(
-                account = account,
-                token = token,
-                url = originalUrl
-              )
-            } else {
+            is LSHTTPResponseStatus.Responded.Error -> {
+              if (status.properties.status == HttpURLConnection.HTTP_NOT_FOUND) {
+                logger.error(
```

Comment:
I don't think this should be at `error` severity. It'll cause a non-fatal event to be logged to Crashlytics. This is probably just `debug`.